source/docs/xds.md

```diff
@@ -0,0 +1,44 @@
+# Envoy xDS subsystem
+
+## Overview
+
+Envoy implements the [xDS
+protocol](https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol). This document tracks
+some implementation details.
+
+An overview of the architecture is provided in the diagram below:
+
+![xDS implementation architecture overview](xDS_code_diagram.png)
+
+Note that this is currently somewhat idealized, with Envoy having two gRPC mux implementations,
+`GrpcMuxImpl` and `NewGrpxMuxImpl`, while we complete migration to a single muxer, see
+https://github.com/envoyproxy/envoy/issues/11477.
+
+The key components are:
+* `GrpcMuxImpl` which manages the xDS transport and multiplexing one or more subscriptions on a
+  single gRPC stream. The muxer is responsible for managing the wire level concerns via
```

Comment:
For non-ADS, it's always one subscription per stream, right? Can you clarify the ADS vs. non-ADS case here a bit?